app/process_proposal.go

```diff
@@ -67,6 +67,9 @@ func (app *App) ProcessProposal(req abci.RequestProcessProposal) (resp abci.Resp
 			tx = blobTx.Tx
 		}
 
+		// todo: uncomment once we're sure this isn't consensus breaking
+		// sdkCtx = sdkCtx.WithTxBytes(tx)
```

Comment:
sorry for one more review @rootulp 

I commented this out as I think there are scenarios where this could be consensus breaking and this is not explicitly tested. It should likely be protected by an if statement for v3

Reply:
cc @ninabarbakadze

Reply:
hmm this PR targets main so we can technically include consensus breaking things b/c main is the basis for v3.0.0.

IIUC we just can't backport this to v2.x

Reply:
do we want to remove this for now? and after this is backported to v2 we'll have it for main with version gates